v2/spanner-common/src/main/java/com/google/cloud/teleport/v2/spanner/migrations/schema/ISchemaOverridesParser.java

```diff
@@ -0,0 +1,40 @@
+/*
+ * Copyright (C) 2024 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+package com.google.cloud.teleport.v2.spanner.migrations.schema;
+
+import org.apache.commons.lang3.tuple.Pair;
+
+public interface ISchemaOverridesParser {
+
+  /**
+   * Gets the spanner table name given the source table name, or source table name if no override is
+   * configured.
+   *
+   * @param sourceTableName The source table name
+   * @return The overridden spanner table name
+   */
+  String getTableOverrideOrDefault(String sourceTableName);
+
+  /**
+   * Gets the spanner column name given the source table name, or the source column name if override
+   * is configured.
+   *
+   * @param sourceTableName the source table name for which column name is overridden
+   * @param sourceColumnName the source column name being overridden
+   * @return A pair of spannerTableName and spannerColumnName
+   */
+  Pair<String, String> getColumnOverrideOrDefault(String sourceTableName, String sourceColumnName);
```

Comment:
Similarly, this would just be getSpannerColumn name - it will return the Spanner column name only if the table name and column name are available in the overrrides. If the tuple is not available, then just return the original column name.

An alternative would be to rename this method to getColumnOverride.
If no override is configured for this tuple, just return null. I think a check for this is being done at the caller for this in any case, so we're not really reducing any complexity by having the or clause here.

Reply:
These type of methods are quite common in Java. My implementation here is just an extension of an already existing concept - 

https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/util/Map.html#getOrDefault(java.lang.Object,V)

I am not particular about the naming here though, so if you think it is confusion, I can change it.

Reply:
Updated the name.